File: python/sfo_10-ii_frog_jump_s1.py
'''
File: sfo_10-ii_frog_jump_s1.py
Created Time: 2021-12-09
Author: Krahets (krahets@163.com)
'''

from include import *

# ===== Solution Code =====
class Solution:
    def numWays(self, n: int) -> int:
        a, b = 1, 1
        for _ in range(n):
            a, b = b, (a + b) % 1000000007
        return a

# ======= Test Case =======
n = 7
# ====== Driver Code ======
slt = Solution()
res = slt.numWays(n)
print(res)

File: cpp/sfo_10ii_frog_jump_s1/sfo_10ii_frog_jump_s1.cpp
/*
* File: sfo_10ii_frog_jump_s1.cpp
* Created Time: 2021-12-09
* Author: Krahets (krahets@163.com)
*/

#include "../include/include.hpp"

// ===== Solution Code =====
class Solution {
public:
    int numWays(int n) {
        int a = 1, b = 1, sum;
        for(int i = 0; i < n; i++){
            sum = (a + b) % 1000000007;
            a = b;
            b = sum;
        }
        return a;
    }
};

int main() {
    // ======= Test Case =======
    int n = 7;
    // ====== Driver Code ======
    Solution* slt = new Solution();
    int res = slt->numWays(n);
    cout << res << endl;
    
    return 0;
}
